x-pack/test/functional/page_objects/security_page.ts

```diff
@@ -400,104 +397,78 @@ export function SecurityPageProvider({ getService, getPageObjects }: FtrProvider
-                await button.click();
-
-                const spaceSelector = await testSubjects.find('spaceSelectorComboBox');
-                await spaceSelector.click();
-
-                const globalSpaceOption = await find.byCssSelector(`#spaceOption_\\*`);
-                await globalSpaceOption.click();
-
-                const basePrivilegeSelector = await testSubjects.find('basePrivilegeComboBox');
-                await basePrivilegeSelector.click();
-
-                const privilegeOption = await find.byCssSelector(`#basePrivilege_${privilegeName}`);
-                await privilegeOption.click();
-
-                const createPrivilegeButton = await testSubjects.find('createSpacePrivilegeButton');
-                await createPrivilegeButton.click();
-              }
-            }
-          })
-
-          .then(function () {
-            function addPrivilege(privileges: string[]) {
-              return privileges.reduce(function (promise: Promise<any>, privilegeName: string) {
-                // We have to use non-test-subject selectors because this markup is generated by ui-select.
-                return promise
-                  .then(() => self.addPrivilegeToRole(privilegeName))
-                  .then(() => PageObjects.common.sleep(250));
-              }, Promise.resolve());
-            }
-            return addPrivilege(roleObj.elasticsearch.indices[0].privileges);
-          })
-          // clicking the Granted fields and removing the asterix
-          .then(async function () {
-            function addGrantedField(field: string[]) {
-              return field.reduce(function (promise: Promise<any>, fieldName: string) {
-                return promise
-                  .then(function () {
-                    return find.setValue('[data-test-subj="fieldInput0"] input', fieldName + '\n');
-                  })
-                  .then(function () {
-                    return PageObjects.common.sleep(1000);
-                  });
-              }, Promise.resolve());
-            }
-
-            if (roleObj.elasticsearch.indices[0].field_security) {
-              // Toggle FLS switch
-              await testSubjects.click('restrictFieldsQuery0');
-
-              // have to remove the '*'
-              return find
-                .clickByCssSelector(
-                  'div[data-test-subj="fieldInput0"] [title="Remove * from selection in this group"] svg.euiIcon'
-                )
-                .then(function () {
-                  return addGrantedField(roleObj.elasticsearch.indices[0].field_security!.grant!);
-                });
-            }
-          }) // clicking save button
-          .then(async () => {
-            log.debug('click save button');
-            await testSubjects.click('roleFormSaveButton');
-          })
-          .then(function () {
-            return PageObjects.common.sleep(5000);
-          })
-      );
+      await this.clickNewRole();
```

Comment:
This code is largely similar to what we used to have, but I removed the promise chain in favor of a series of async functions. I find it easier to read and reason about. This also uses the `comboBox` service below, and removes the fixed wait of `5000ms`